account_cash_invoice/wizard/cash_pay_invoice.py

```diff
@@ -0,0 +1,108 @@
+        active_ids = self.env.context.get("active_ids")
+        journal = self.env["account.journal"].browse(active_ids)
+        return journal.company_id
+
+    def _default_currency(self):
+        active_ids = self.env.context.get("active_ids")
+        journal = self.env["account.journal"].browse(active_ids)
+        return journal.currency_id or journal.company_id.currency_id
+
+    invoice_id = fields.Many2one(
+        comodel_name="account.move",
+        string="Invoice",
+        required=True,
+    )
+    name = fields.Char(related="invoice_id.name", readonly=True)
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        default=lambda self: self._default_company(),
+        required=True,
+        readonly=True,
+    )
+    currency_id = fields.Many2one(
+        comodel_name="res.currency",
+        default=lambda self: self._default_currency(),
+        required=True,
+        readonly=True,
+    )
+    journal_id = fields.Many2one(
+        comodel_name="account.journal",
+        required=True,
+        readonly=True,
+        default=lambda self: self.env.context.get("active_ids")[0],
+        string="Journal",
+    )
+    amount = fields.Monetary()
+    invoice_type = fields.Selection(
+        [
+            ("customer", "Customer"),
+            ("vendor", "Vendor"),
+        ],
+    )
+    invoice_domain = fields.Binary(compute="_compute_invoice_domain")
+
+    @api.depends("company_id", "currency_id", "invoice_type")
+    def _compute_invoice_domain(self):
+        for wizard in self:
+            invoice_domain = [
+                ("company_id", "=", wizard.company_id.id),
+                ("currency_id", "=", wizard.currency_id.id),
+                ("state", "=", "posted"),
+                ("payment_state", "!=", "paid"),
+            ]
+            if wizard.invoice_type:
+                move_type = (
+                    ["out_invoice", "in_refund"]
+                    if wizard.invoice_type == "customer"
+                    else ["in_invoice", "out_refund"]
+                )
+                invoice_domain.append(("move_type", "in", move_type))
+            else:
+                # to avoid select invoice if invoice_type is not set
+                invoice_domain.append(("move_type", "in", []))
+            wizard.invoice_domain = invoice_domain
+
+    @api.onchange("invoice_type")
+    def _onchange_invoice_type(self):
+        self.invoice_id = False
+
+    @api.onchange("invoice_id")
+    def _onchange_invoice(self):
+        if self.invoice_id:
+            self.amount = self.invoice_id.amount_residual_signed
+
+    def action_pay_invoice(self):
+        BankStatementLine = self.env["account.bank.statement.line"]
+        statement_line_vals = self._prepare_values_for_statement_line(self)
+        BankStatementLine.create(statement_line_vals)
+        return {"type": "ir.actions.act_window_close"}
+
+    def _prepare_values_for_statement_line(self, record):
```

Comment:
Convention for method names is:

```suggestion
    def _prepare_statement_line_vals(self):
```

And `record` is not needed. And if `record` would be another model record, the prepare method should be there in such model, not here, but it's not the case.